lib/bolt/pal.rb

```diff
@@ -145,6 +145,13 @@ def in_bolt_compiler
             end
           rescue Bolt::Error => e
             e
+          rescue Puppet::DataBinding::LookupError => e
```

Comment:
It'd be nice if we also caught `Unknown variable $trusted / $server_facts / $settings::` here - while I highly doubt anyone is using them (and if they are they're surely misusing them) it'd be crazy-making to have a working plan suddenly start failing with no clue as to why.

Reply:
https://github.com/puppetlabs/bolt/pull/1677/files#diff-a0f737621da3fb6ead05b549796d7406R156-R159

Not really sure what the best phrasing for the message is, but that will catch any errors thrown by plans that were previously accessing those variables.